client.go

```diff
@@ -913,15 +931,27 @@ func (f *File) Read(b []byte) (int, error) {
 // maximise throughput for transferring the entire file (especially
 // over high latency links).
 func (f *File) WriteTo(w io.Writer) (int64, error) {
-	fi, err := f.c.Stat(f.path)
-	if err != nil {
-		return 0, err
+	var err error
+	fileSize := uint64(0)
+	if f.c.useFstat {
+		var fileStat *FileStat
+		if fileStat, err = f.c.fstat(f.handle); err != nil {
```

Comment:
Because you’re going to continue using the `fileStat` variable, it’s way better to use the `:=` short declaration and initialization outside, rather than using this very unwieldy declaration and precondition.

So:
```
fs, err := f.c.fstat(f.handle)
if err != nil {
  return 0, err
}
fileSize = fs.Size // casting is unnecessary, as the types already match
```

Basically, the overriding interest of avoiding explicit type declarations outweighs the interest in putting things into an initializing precondition.